### ISSUE TITLE
slow when ordering groups in the object panel 

### ISSUE THREAD

**Issue by ponderboy** [TheMightyFingers/mightyeditor#36]
ME runs really slow when I arrange the order of groups within the objects panel. 

I can drag a group within one group and it takes about fifteen seconds before it shows the update. I thought maybe turning off visibility of the group and then dragging things in order might work better, but still it's slow.

**Comment by stauzs** [comment]
I cannot reproduce. 
* How many groups does your project have?
* Can you repeat it 100% of the time (can it be related to the connection to the server)
* Does turning off webGL makes any difference? (MyMighty -> Preferences -> webGl - ME reload required to change Phaser's renderer mode)

**Comment by ponderboy** [comment]
* I have four main groups each containing between two to six groups, which themself contain up to ten or more groups that might contain groups too
* yes it happens to me always since using ME
* There is no "Preferences" under myMighty

I'm aware of not using ME or Phaser in the typical game development workflow. I also have more of a design background. The game I'm doing is more like a "click through screens" sort of thing. Quite UI heavy with bigger graphics and so on. 

I also have this problems when copying or duplicating groups. 
Sometimes shift selecting groups in the object panel is not working at the first attempt and so on...

For me I was blaming the "newness" of ME as well as that it is running in the browser. 
I have to admit, I would not consider it production ready. I can handle all those little drawbacks as it saves me quite a lot of coding. But I can not imagine it being used in an agency for example. At least if there are no people proficient in using ME. 

I use a nearly three year old laptop and the graphic card is not the best. ( i7-2670 QM CPU @ 2.2 Ghz and 8GB of RAM, Windows 7, GeForce GT 540M). If that is of any help or interest. 
* I have four main groups each containing between two to six groups, which themself contain up to ten or more groups that might contain groups too
* yes it happens to me always since using ME
* There is no "Preferences" under myMighty

I'm aware of not using ME or Phaser in the typical game development workflow. I also have more of a design background. The game I'm doing is more like a "click through screens" sort of thing. Quite UI heavy with bigger graphics and so on. 

I also have this problems when copying or duplicating groups. 
Sometimes shift selecting groups in the object panel is not working at the first attempt and so on...

For me I was blaming the "newness" of ME as well as that it is running in the browser. 
I have to admit, I would not consider it production ready. I can handle all those little drawbacks as it saves me quite a lot of coding. But I can not imagine it being used in an agency for example. At least if there are no people proficient in using ME. 

I use a nearly three year old laptop and the graphic card is not the best. ( i7-2670 QM CPU @ 2.2 Ghz and 8GB of RAM, Windows 7, GeForce GT 540M). If that is of any help or interest.

**Comment by stauzs** [comment]
* Sorry: webgl flag is located under MyMighty -> Personalise -> webGl ( set it to 0 and reload browser )

Actually after creating approx 10 groups containing 4-7 objects each - there is noticeable slow down. (1-2 seconds).

Does it started to happen only after the last update for you? 

P.S. I have similar setup laptop ( only chrome is using built-in video card intel (HD 4000) - probably slower than geforce

**Comment by ponderboy** [comment]
Found the webgl flag. But it didn't change much. Copying seemed to work better now. (Although I can't shift paste it in the selected group, but that's another story) 

No, it did not start only after the last update. I was experiencing this ever since my project grew bigger.

**Comment by stauzs** [comment]
I optimized a little bit sorting part - now sorting should be faster ( update is already on mightyeditor,mightyfingers.com ) - I'll push it to the github in the next hour

**Comment by ponderboy** [comment]
Mhm, have to say, it doesn't feel faster for me.

**Comment by stauzs** [comment]
can you reload ME with Ctrl+F5 or Shift+F5? To force re-download scripts?

**Comment by ponderboy** [comment]
well, that did not help.

**Comment by ponderboy** [comment]
I cleaned out the whole history and cache and all in Chrome and now dragging works good. 
Also copy pasting works fast now. Only it never gets pasted within the same group, just on top of all in the objects panel. 
Great!
